spring-boot/src/main/java/org/springframework/boot/web/support/ErrorPageFilter.java

```diff
@@ -223,12 +223,12 @@ private String getErrorPath(Class<?> type) {
 		if (this.subtypes.containsKey(type)) {
 			return this.exceptions.get(this.subtypes.get(type));
 		}
-		Class<?> subtype = type;
-		while (subtype != Object.class) {
-			subtype = subtype.getSuperclass();
-			if (this.exceptions.containsKey(subtype)) {
-				this.subtypes.put(subtype, type);
-				return this.exceptions.get(subtype);
+		Class<?> supertype = type;
+		while (supertype != Object.class) {
+			supertype = supertype.getSuperclass();
+			if (this.exceptions.containsKey(supertype)) {
+				this.subtypes.put(type, supertype);
```

Comment:
Hi! Yes It was intentional as I've mentioned in the first comment to this PR. I also changed variable name from `subtype` to `supertype` because I suspect it was the reason for initial mistake (_swapped put arguments_) since it was suggesting different relation between arguments.